### PR TITLE
Try to enable ScalaJS 1.0.0 support #78

### PR DESCRIPTION
**Pull request by zy4**
I tried to update the dependencies to enable the support but I am not very familiar with your test setup. At least `sbt test` seems to fail.

### CODE REVIEW

**Comment by zy4** [comment]
Except `sbt ++$TRAVIS_SCALA_VERSION scalastyle` everything seem to pass.

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,9 +1,11 @@
-addSbtPlugin("org.scala-js" % "sbt-scalajs" % "0.6.15")
+val scalaJSVersion = Option(System.getenv("SCALAJS_VERSION")).getOrElse("0.6.19")
```
Comment (hmil) [inline project/plugins.sbt:1]: Why are you pulling the version from the environment ? Can't we just set it to `1.0.0-M2` or something?

Comment (sjrd) [reply project/plugins.sbt:1]: That's the recommended way to cross-build across Scala.js 0.6.x and 1.x: https://www.scala-js.org/news/2017/11/29/announcing-scalajs-1.0.0-M2/

**Review by sjrd — CHANGES_REQUESTED** [review]
This PR is crucially missing changes in `.travis.yml` to actually build and test with 1.x on the CI.

**Comment by hmil** [comment]
I tried fixing the issues in https://github.com/hmil/RosHTTP/tree/zy4-scalajs1.0.0 but I'm still getting errors in build.sbt

**Comment by sjrd** [comment]
```
/home/travis/build/hmil/RosHTTP/build.sbt:52: error: not found: value NodeJSEnv
    jsEnv := NodeJSEnv().value
             ^
```
That you can solve simply by removing that line from the build. `NodeJSEnv` has been the default since 0.6.13 or so. The equivalent now would be:
```scala
jsEnv := new org.scalajs.jsenv.nodejs.NodeJSEnv()
```
The explicit dependency on `scalajs-env-nodejs` in `project/plugins.sbt` is redundant, and should be removed to allow the 0.6.x build to work.

**Pull request closed without merging**